### ISSUE TITLE
DNS, IP and Caddy configuration synchronization issue during AWS ECS Pipeline deployments

### ISSUE THREAD

**Issue by RaulJMCD** [caddyserver/caddy#6748]
### Issue Description
During automated deployments in AWS ECS, we're experiencing a synchronization issue between DNS updates and Caddy's proxy configuration. When a new task is deployed, it gets a new IP, but there's a gap where Caddy's proxy configuration doesn't match the new DNS record.

### Current Setup
- Running Caddy 2.8.4 in AWS ECS Fargate
- Using Route53 for DNS management
- Caddy and the API run in the same task definition but in different containers
- Using a script before API service start to update Route53 when new task is deployed
- API service starts first and caddy starts when API is healthy

### Current Workflow
1. ECS deploys a new task
2. Script updates Route53 A record with the new task's IP
3. API service starts
4. Caddy is configured with a static proxy setup
5. After CI Pipeline deployment DNS is not solved correctly and navigation doesnt happens
6. I manually stops task and when it automatically starts again it always works properly
### Current Configuration
```hcl
reverse_proxy 127.0.0.1:3000 {
    health_uri /api/health
    health_interval 60s
    max_fails 10
}

### What I've Tried
Using sleep delays before starting Caddy
Implementing health checks
Update script already check actively in route53 the INSYNC state 
**Manual task deletion (which fix the issue)**

### Question
What's the recommended approach to handle dynamic IP updates in Caddy when running in a container orchestration environment? Is there a way to make Caddy automatically detect and adapt to IP changes, or should we implement a dynamic configuration update mechanism?

### Additional Contex
The issue only occurs during automated deployments
Manual restarts of the task resolve the problem
Looking for a solution that maintains zero-downtime deployments

Any guidance or best practices for handling this scenario would be greatly appreciated.
